### ISSUE TITLE
Opening Blood Hunter page on D&D Beyond begins repeatedly refreshing.

### ISSUE THREAD

**Issue by ya-hs** [kakaroto/Beyond20#851]
When the extension is enabled on my browser and I open the Blood Hunter class page, my Chrome tab refreshes rapidly before triggering the anti-bot system built into the website. This issue does not occur when the extension is disabled.

**Comment by kakaroto** [comment]
I see it. Confirmed.
Thanks!

**Comment by Aeristoka** [comment]
Is this another "people put crap in the chat below and Beyond20 is trying desperately to parse it" thing?

**Comment by kakaroto** [comment]
Good catch! yep, that's what it is. There's a custom b20 dice in a comment, so when the page loads, Beyond20 thinks the extension was just enabled on a page that was already parsed, so it reloads it to clear it... It does the same for monster and spell pages too... 
```js

function documentLoaded(settings) {
    character = new SourceBookCharacter(settings);
    if (isRollButtonAdded() || isCustomRollIconsAdded()) {
        chrome.runtime.sendMessage({ "action": "reload-me" });
    } else {
```

Will need to give those two functions (`isRollButtonAdded` and `isCustomRollIconsAdded`) a selector for where to look, so it only checks the content and ignores the comments.

**Comment by kakaroto** [comment]
it's slightly more complex than expected because of how the html is set up in the page. I can't give it a selector to say "check this div" which excludes comments, because they're all siblings and part of the main primary-content div.
The issue doesn't happen anymore though because it appears that comment has been fixed/deleted. I assume someone might have reported it to the ddb team.

**Comment by kakaroto** [comment]
Closing this as it's fixed, but the important part is that issue #889 has been fixed which means it shouldn't happen again in the future either 👍
